src/ContextualPopup/ContextualPopup.js

```diff
@@ -568,10 +606,10 @@ module.exports = kind(
 		// show a transparent scrim for modal popups if
 		// {@link module:moonstone/ContextualPopup~ContextualPopup#scrimWhenModal} is `true`, else show a
 		// regular scrim.
-		if (this.modal && this.scrimWhenModal) {
+		if (this.modal && this.scrimWhenModal && !this.scrim) {
```

Comment:
Hiding the previous scrim makes sense to me. I'm on the fence about how much work we want to do to fix up some of the misguided logic, as you mentioned this also exists in `moonstone/Popup`. My comment above is along those lines; let's fix things where we can, but if it involves some major refactoring I think it would be best to hold off.